pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -455,16 +455,15 @@ def _choose_accelerator(self) -> str:
         return "cpu"
 
     def _set_parallel_devices_and_init_accelerator(self) -> None:
-        # TODO add device availability check
+        ACCELERATORS = {
+            "cpu": CPUAccelerator,
+            "gpu": GPUAccelerator,
+            "tpu": TPUAccelerator,
+            "ipu": IPUAccelerator,
+        }
```

Comment:
currently this duplicates the name implementation of each accelerator class here. this is a risk in case the name ever changes (unlikely to happen, but still). this would've protected against that:
```py
acc_cls = (CPUAccelerator, GPUAccelerator, TPUAccelerator, IPUAccelerator)
ACCELERATORS = {acc_cls.name(): acc_cls for acc in acc_cls}
```

Reply:
I think this will be removed here: https://github.com/PyTorchLightning/pytorch-lightning/pull/12180